Content.Shared/Damage/DamageGroupPrototype.cs

```diff
@@ -8,34 +8,69 @@
 namespace Content.Shared.Damage
 {
     /// <summary>
-    /// A Group of DamageTypes.
+    /// A Group of <see cref="DamageTypePrototype"/>s .
     /// </summary>
     [Prototype("damageGroup")]
     [Serializable, NetSerializable]
     public class DamageGroupPrototype : IPrototype, ISerializationHooks
     {
-        [Dependency] private IPrototypeManager _prototypeManager = default!;
+        private IPrototypeManager _prototypeManager = default!;
 
         [DataField("id", required: true)] public string ID { get; } = default!;
 
         [DataField("damageTypes", required: true)]
-        public List<string> TypeIds { get; } = default!;
+        public List<string> TypeIDs { get; } = default!;
 
+        public HashSet<DamageTypePrototype> DamageTypes { get; } = new();
+
+
+        // Create set of damage types
         void ISerializationHooks.AfterDeserialization()
         {
             _prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+
+            foreach (var typeID in TypeIDs)
+            {
+                DamageTypes.Add(_prototypeManager.Index<DamageTypePrototype>(typeID));
+            }
+
         }
 
-        public IEnumerable<DamageTypePrototype> DamageTypes
+    // TODO QUESTION where shold this go? a function like this was previously in DamageableComponent, but it seems more
```

Comment:
Does a function like this already exist somewhere? Should this function be here?

Reply:
i'd prefer if it was in DamageableComponent